crates/miden-lib/asm/kernels/transaction/lib/asset_vault.masm

```diff
@@ -129,6 +116,41 @@ export.peek_balance
     # => [balance]
 end
 
+#! Returns the fungible asset associated with a faucet_id.
+#!
+#! Inputs:  [faucet_id_prefix, faucet_id_suffix, vault_root_ptr]
+#! Outputs: [ASSET]
+#!
+#! Where:
+#! - vault_root_ptr is a pointer to the memory location at which the vault root is stored.
+#! - faucet_id_{prefix, suffix} are the prefix and suffix felts of the faucet id of the fungible
+#!   asset of interest.
+#! - ASSET is the fungible asset of interest.
+#!
+#! Panics if:
+#! - the asset is not a fungible asset.
+export.get_fungible_asset
```

Comment:
Related to one of the previous comments: could we not just pass the pointer to the initial account vault into this procedure and get what we need (i.e., no need to do anything with account delta)?